src/openssl.c

```diff
@@ -1487,15 +1487,17 @@ static const char *auxL_pusherror(lua_State *L, int error, const char *fun) {
 			file = path;
 		}
 
-		ERR_clear_error();
-
 		ERR_error_string_n(code, txt, sizeof txt);
 
 		if (fun) {
-			return lua_pushfstring(L, "%s: %s:%d:%s", fun, file, line, txt);
+			lua_pushfstring(L, "%s: %s:%d:%s", fun, file, line, txt);
 		} else {
-			return lua_pushfstring(L, "%s:%d:%s", file, line, txt);
+			lua_pushfstring(L, "%s:%d:%s", file, line, txt);
```

Comment:
`lua_pushfstring` can fail (i.e. `longjmp` out). We need to clear the OpenSSL error before calling it.

Reply:
I'd rather take a potential longjmp (due to memory allocation failure) over consistently using released memory (the current situation).

Nevertheless, I'm happy to amend the patch. But what do you suggest? Allocating a temporary buffer to hold the string also has the potential to fail if lua_pushfstring() may fail. Then what?

I haven't dug deeply into the OpenSSL error handling API docs, but as far as I can tell, if we don't clear the error it just remains on the stack, it doesn't actually "leak".

Reply:
We already have the temporary (stack-allocated) buffer `txt`. `ERR_error_string_n` copies into that.
So the patch really only needs to swap the order of `ERR_clear_error` and `ERR_error_string_n`